File: projects/pinstripe/packages/@pinstripe/core/lib/commands/list_views.js

import chalk from 'chalk';

import { Command } from '../command.js';
import { View } from '../view.js';

Command.register('list-views').define(dsl => dsl.props({
    run(){
        console.log('');
        console.log('The following views are available:');
        console.log('');
        Object.keys(View.classes).sort().forEach(viewName => {
            console.log(`  * ${chalk.green(viewName)}`);
        });
        console.log('');
    }
}));

File: projects/slick/lib/slick/commands/list_views.rb

require "slick/command"

Slick::Command.register("list-views").define_method "run" do

    puts ""
    puts "The following views are available:"
    puts ""
    Slick::View.registered_classes.keys.sort.each do |name|
        puts "  * #{name.green}"
    end
    puts ""

end


